src/main/java/soconnect/logic/autocomplete/Autocomplete.java

```diff
@@ -1,29 +1,71 @@
 
     /**
-     * Updates the set of unique names whenever a command is executed.
+     * Generates a list of autocomplete entries that starts with the given user input.
      *
-     * @param uniqueNames The updated set of unique names.
+     * @param userInput The input from the user.
+     * @return A list of matching autocomplete entries.
      */
-    void updateUniqueNames(TreeSet<String> uniqueNames);
+    List<String> getAutocompleteEntries(String userInput);
 
     /**
-     * Generates a list of autocomplete entries that starts with the given searchValue.
+     * Gets a valid argument if the given {@code userInput} is a search command input,
+     * otherwise returns an invalid argument. It determines whether the {@code userInput} is a
+     * search command input by checking if the {@code userInput} has the command word "search".
      *
-     * @param searchValue The value to be matched with.
-     * @return A list of matching autocomplete entries.
+     * @param userInput The input from the user.
+     * @return A valid argument if the {@code userInput} is a search command input, otherwise an invalid argument.
+     */
+    String getSearchCommandArguments(String userInput);
+
+    /**
+     * Gets the last prefix argument string from the given {@code argsString} if the prefix and argument are present,
+     * otherwise returns an invalid prefix argument string.
+     *   <br>examples:<pre>
+     *       If{@code argsString} = "and n/alex p/900", {@code prefix} = "n/" and "p/", this method returns
+     *       "p/900" as it is the last prefix with an argument.
+     *       If {@code argsString} = "and n/alex p/", {@code prefix} = "n/" and "p/", this method returns
+     *       and invalid prefix argument as the last prefix "p/" does not have an argument.
+     *       If {@code argsString} = "and n/alex p/", {@code prefix} = "e/" and "a/", this method returns
+     *       and invalid prefix argument as there is no prefix in the {@code argsString}.
+     *   </pre>
+     *
+     * @param argsString The arguments string of the form: {@code preamble <prefix>value <prefix>value ...}
+     * @param prefixes Prefixes to tokenize the arguments string with.
+     * @return The string of the last prefix with argument if they are present, otherwise an invalid prefix argument.
+     */
+    String getLastPrefixArgument(String argsString, Prefix... prefixes);
+
+    /**
+     * Updates the filter of the filtered person list based on the preamble in the given {@code argsString}.
+     *
+     * @param argsString Arguments string of the form: {@code preamble <prefix>value <prefix>value...}
+     */
+    void updateFilteredPersonList(String argsString);
+
+    /**
+     * Generates a list of autocomplete entries.
+     *
+     * @param argsWithoutLastPrefixArgument The arguments string of the
+     *                                      form:{@code preamble <prefix>value <prefix>value ...}
```

Comment:
```suggestion
     *                                      form:{@code preamble <prefix>value <prefix>value ...}.
```

Reply:
From the AB3 documentation, they didn't include a full stop after the ellipses so I just follow. 

![image](https://user-images.githubusercontent.com/56215606/197379403-5a975ea2-b998-41f0-b37e-44bfd98c9c75.png)

I tried adding the full stop and it looks like this which is quite ugly to me so I went to search about it.

Based on this website https://www.stylemanual.gov.au/grammar-punctuation-and-conventions/punctuation-and-capitalisation/ellipses#:~:text=Don't%20use%20a%20full,or%20semicolon%20after%20an%20ellipsis, we do not have to include a full stop after it.